packages/shared/src/components/post/RelatedPostsWidget.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { widgetClasses } from '../widgets/common';
+import { InfinitePaginationActions } from '../pagination';
+import { ElementPlaceholder } from '../ElementPlaceholder';
+import {
+  Post,
+  PostRelationType,
+  RELATED_POSTS_PER_PAGE_DEFAULT,
+} from '../../graphql/posts';
+import { SourceAvatar } from '../profile/source';
+import PostMetadata from '../cards/PostMetadata';
+import { CardLink } from '../cards/Card';
+import { useRelatedPosts } from '../../hooks/post';
+
+export type RelatedPostsWidgetProps = {
+  className?: string;
+  post: Post;
+  perPage?: number;
+  relationType: PostRelationType;
+};
+
+export const RelatedPostsWidget = ({
+  className,
+  post,
+  perPage = RELATED_POSTS_PER_PAGE_DEFAULT,
+  relationType,
+}: RelatedPostsWidgetProps): ReactElement => {
+  const {
+    relatedPosts,
+    isLoading,
+    isFetchingNextPage,
+    hasNextPage,
+    fetchNextPage,
+  } = useRelatedPosts({
+    postId: post.id,
+    relationType,
+    perPage,
+  });
+  const [page, setPage] = useState(0);
+  const relatedPostPage = relatedPosts?.pages[page];
+  const hasRelatedPosts = isLoading || !!relatedPostPage?.edges.length;
+
+  if (!hasRelatedPosts) {
+    return null;
+  }
+
+  return (
+    <div className={classNames(className, 'flex flex-col', widgetClasses)}>
+      <div
+        className={classNames(
+          'rounded-16 cursor-pointer laptop:cursor-auto flex justify-between items-center py-1.5 laptop:py-4 px-4 laptop:!pb-0 rounded-t-16',
+        )}
+      >
+        <p className="font-bold typo-callout text-theme-label-quaternary laptop:text-theme-label-quaternary">
+          {post.numCollectionSources} sources
+        </p>
+      </div>
+      <div
+        className={classNames('flex pt-2 pb-0 laptop:min-h-[14rem] flex-col')}
```

Comment:
Should not be needed here as well.
```suggestion
        className="flex pt-2 pb-0 laptop:min-h-[14rem] flex-col"
```

Reply:
Good catch.